File: android/src/main/kotlin/com/tsitser/background_http_plugin/domain/usecase/DeleteRequestUseCase.kt
package com.tsitser.background_http_plugin.domain.usecase

import com.tsitser.background_http_plugin.domain.repository.TaskRepository

/**
 * Use case для удаления задачи
 */
class DeleteRequestUseCase(
    private val repository: TaskRepository
) {
    suspend operator fun invoke(requestId: String): Boolean? {
        return repository.deleteTask(requestId)
    }
}

File: ios/Classes/Domain/UseCase/DeleteRequestUseCase.swift
import Foundation

/// Use case for deleting a task
struct DeleteRequestUseCase {
    private let repository: TaskRepository
    
    init(repository: TaskRepository) {
        self.repository = repository
    }
    
    func execute(requestId: String) async throws -> Bool? {
        return try await repository.deleteTask(requestId: requestId)
    }
}

